src/client/game.cpp

```diff
@@ -3012,51 +3008,50 @@ void Game::processPlayerInteraction(f32 dtime, bool show_hud, bool show_debug)
 			client->setCrack(-1, v3s16(0, 0, 0));
 			runData.dig_time = 0.0;
 		}
-	} else if (runData.dig_instantly && input->getLeftReleased()) {
-		// Remove e.g. torches faster when clicking instead of holding LMB
+	} else if (runData.dig_instantly && !isKeyDown(KeyType::DIG)) {
```

Comment:
`!isKeyDown` is perhaps not the same as `getLeftReleased`. The former just tells the state while the latter maybe returns whether the key was released once between the steps.
Or maybe they are the same, I don't know.

Reply:
^ Needs consideration.

Reply:
Good catch, thanks! It's indeed `wasKeyDown` that should be used here. `isKeyDown` corresponds to `getLeftState`, and `!isKeyDown` isn't the same as `wasKeyDown`.

Reply:
I understand that `!isKeyDown` isn't the same as `get(Left|Right)Released`. I'll implement an `isKeyReleased` instead.

Reply:
I've added a `wasKeyPressed` which is equivalent to `get(Left|Right)Clicked`. This method is very similar to `wasKeyDown` but unlike the latter, it's state isn't reset immediately after calling but only at the end of `Game::processPlayerInteraction` - this means that this function will return the same value for all `wasKeyPressed` calls in `Game::processPlayerInteraction` in a given iteration.

(Thanks for reminding me to comment on this, @sfan5)

Reply:
so is `!isKeyDown(KeyType::DIG)` the correct equivalent of `getLeftReleased()` then?

Reply:
`getLeftReleased` returns `true` only for one run of `Game::processPlayerInteraction` and is reset (https://github.com/minetest/minetest/pull/7924/files#diff-7f71a92de20504e8a5636b4317ea67b2L3169). So it's indeed different from `!isKeyDown()` which always evaluates to true if the key is up. *But*, the behaviour is the same in this case, as `dig_instantly` and `!isKeyDown` both need to be true for the inner code to run. Once the inner code runs, `dig_instantly` is set to false, and the code never runs again until the key is pressed again. Essentially, `dig_instantly` is already used to keep track of key release events, so `!down` is the same as `released` in this specific context.

Reply:
I've changed `wasKeyPressed` to evaluate to true only when the key was pressed down. So this is equivalent to `isKeyDown`, except that it's state would be reset after one step. I've now added a separate `wasKeyReleased` (which I should've done much earlier) which returns true for one step after a key has been released. I'm still in the process of testing this, but this should fix everything (TM).